Document aggregated minor pod usage (#38223)
DataDog/documentation: content/en/cloud_cost_management/allocation/container_cost_allocation.mdoc.md

{% /if %}

### Aggregated minor pod usage

> In rare cases, many pods run on one node. Datadog compresses the lowest-resource pods into an aggregate. This preserves their combined CPU, memory, and GPU resource-hours and allocated cost. Pod-level tag granularity is reduced for this small portion of spend.

Reviewer: I think "many" pods is underselling the fact that top K is enabled for like top 250? top 500? I guess i'm not sure how anomalous these hundreds of pods on one node are, or if it's more of an anomaly

Author: ```suggestion
In rare cases, a single node can be inundated with many short lived small pods. Datadog compresses the lowest-resource pods into an aggregate. This preserves their combined CPU, memory, and GPU resource-hours and allocated cost. Pod-level tag granularity is reduced for this small portion of spend.
```

What about this wording? Based on my testing this is actually pretty anomalous and rare. And when you account for the fact that we compress only the smallest cheapest pods, the total cost of these pods has been nearly immaterial.

> In rare cases, a single node can be inundated with many short lived small pods. Datadog compresses the lowest-resource pods into an aggregate. This preserves their combined CPU, memory, and GPU resource-hours and allocated cost. Pod-level tag granularity is reduced for this small portion of spend.

Reviewer 2: ```suggestion
In rare cases, a single node runs a large number of short-lived, small pods. Datadog compresses the lowest-resource pods into an aggregate that preserves their combined CPU, memory, and GPU resource-hours and allocated cost. Pod-level tag granularity is reduced for this small portion of spend.
```

> ### Aggregated minor pod usage
> 
> In rare cases, many pods run on one node. Datadog compresses the lowest-resource pods into an aggregate so container cost allocation can complete reliably. This preserves their combined CPU, memory, and GPU resource-hours and allocated cost. Pod-level tag granularity is reduced for this small portion of spend.
> 
> Tag values shared by every compressed pod are retained. Individual values for missing or differing tags are replaced by these context values:
> 
> - `aggregated_pod_kube_cluster_name:<cluster>`
> - `aggregated_pod_kube_node_name:<node>`
> 
> Use `pod_allocation_source:aggregated_minor_usage` to identify these rows.
> 

Author: ```suggestion
### Aggregated minor pod usage

In rare cases, many pods run on one node. Datadog compresses the lowest-resource pods into an aggregate. This preserves their combined CPU, memory, and GPU resource-hours and allocated cost. Pod-level tag granularity is reduced for this small portion of spend.

Individual values for missing or differing tags are replaced by these context values:

- `aggregated_pod_kube_cluster_name:<cluster>`
- `aggregated_pod_kube_node_name:<node>`

Use `pod_allocation_source:aggregated_minor_usage` to identify these rows.

```
